File: frontend/turnero-medico/src/types/Medicamento.ts
// src/types/Medicamento.ts

export interface Laboratorio {
  id: number;
  nombre: string;
}

export interface Medicamento {
  id: number;
  nombre: string;
  descripcion?: string;
  laboratorios: Laboratorio[]; // Siempre llega como lista (puede ser vacía)
}

export interface MedicamentoCreate {
  nombre: string;
  descripcion?: string;
  ids_laboratorios: number[]; // el backend espera SIEMPRE lista, incluso vacía
}

export interface MedicamentoUpdate {
  nombre?: string;
  descripcion?: string;
  ids_laboratorios?: number[];
}

export interface MedicamentoPayload {
  id?: number;
  nombre: string;
  descripcion?: string;
  ids_laboratorios?: number[];
}

File: backend/app/domain/schemas/medicamento.py
from pydantic import BaseModel, ConfigDict
from typing import Optional
from .laboratorio import LaboratorioRead

class MedicamentoBase(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    nombre: str
    descripcion: Optional[str] = None

class MedicamentoCreate(MedicamentoBase):
    ids_laboratorios: list[int] = []

class MedicamentoUpdate(MedicamentoBase):
    nombre: str
    descripcion: Optional[str] = None
    ids_laboratorios: Optional[list[int]] = None

class MedicamentoRead(MedicamentoBase):
    id: int
    laboratorios: list[LaboratorioRead] = []